File: CS/ViewerForm1.cs
using DevExpress.DashboardCommon.ViewerData;
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace RandomFilterExample
{
    public partial class ViewerForm1 : XtraForm {
        public ViewerForm1() {
            InitializeComponent();
            dashboardViewer1.Initialized += OnDashboardViewerInitialized;

        }
        async void OnMouseClick(object sender, MouseEventArgs e) {
            await RandomFilter();
        }
        async Task RandomFilter() {
            string itemName = "choroplethMapDashboardItem1";
            IList<AxisPointTuple> filters = await dashboardViewer1.GetAvailableFilterValuesAsync(itemName);
            Random r = new Random();
            int index = r.Next(0, filters.Count - 1);
            await dashboardViewer1.SetMasterFilterAsync(itemName, filters[index]);
        }

        async void OnDashboardViewerInitialized(object sender, EventArgs e) {
            await RandomFilter();
        }
    }
}

File: VB/ViewerForm1.vb
Imports DevExpress.DashboardCommon.ViewerData
Imports DevExpress.XtraEditors
Imports System
Imports System.Collections.Generic
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace RandomFilterExample
	Partial Public Class ViewerForm1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			AddHandler dashboardViewer1.Initialized, AddressOf OnDashboardViewerInitialized

		End Sub
		Private Async Overloads Sub OnMouseClick(ByVal sender As Object, ByVal e As MouseEventArgs) Handles simpleButton1.MouseClick
			Await RandomFilter()
		End Sub
		Private Async Function RandomFilter() As Task
			Dim itemName As String = "choroplethMapDashboardItem1"
			Dim filters As IList(Of AxisPointTuple) = Await dashboardViewer1.GetAvailableFilterValuesAsync(itemName)
			Dim r As New Random()
			Dim index As Integer = r.Next(0, filters.Count - 1)
			Await dashboardViewer1.SetMasterFilterAsync(itemName, filters(index))
		End Function

		Private Async Sub OnDashboardViewerInitialized(ByVal sender As Object, ByVal e As EventArgs)
			Await RandomFilter()
		End Sub
	End Class
End Namespace
